kernel/e1000/src/regs.rs

```diff
@@ -1,3 +1,169 @@
+#[repr(C)]
+pub struct E1000RxRegisters {
+    _padding6:                      [u8; 2048],             // 0x2000 - 0x27FF
+
+    pub rx_regs:                    RegistersRx,            // 0x2800    
+    _padding7a:                     [u8; 1024],             // 0x281C - 0x2FFF
+    _padding7b:                     [u8; 512],
+    _padding7c:                     [u8; 256],
+    _padding7d:                     [u8; 128],  
+    _padding7e:                     [u8; 64],
+    _padding7f:                     [u8; 36],  
+} // 1 4KiB page
+
+const_assert_eq!(core::mem::size_of::<E1000RxRegisters>(), 4096);
+
+
+/// The layout in memory of e1000 transmit registers. 
+/// 
+/// Note: the weird padding is a limitation of using the `zerocopy::FromBytes` trait,
+/// which in the absence of const generics, only implements its traits for arrays of [T: N] 
+/// where N is a power of two or is less than 64.
+#[derive(FromBytes)]
+#[repr(C)]
+pub struct E1000TxRegisters {
+    _padding8:                      [u8; 2048],             // 0x3000 - 0x37FF
+
+    pub tx_regs:                    RegistersTx,            // 0x3800
+    _padding9a:                     [u8; 1024],             // 0x381C - 0x3FFF
+    _padding9b:                     [u8; 512],
+    _padding9c:                     [u8; 256],
+    _padding9d:                     [u8; 128],  
+    _padding9e:                     [u8; 64],
+    _padding9f:                     [u8; 36],  
+} // 1 4KiB page
+
+const_assert_eq!(core::mem::size_of::<E1000TxRegisters>(), 4096);
+
+
+/// The layout in memory of e1000 MAC address registers. 
+/// 
+/// Note: the weird padding is a limitation of using the `zerocopy::FromBytes` trait,
+/// which in the absence of const generics, only implements its traits for arrays of [T: N] 
+/// where N is a power of two or is less than 64.
+#[derive(FromBytes)]
+#[repr(C)]
+pub struct E1000MacRegisters {
+    _padding10a:                    [u8; 4096],             // 0x4000 - 0x53FF
+    _padding10b:                    [u8; 1024],             
+    
+    /// The lower (least significant) 32 bits of the NIC's MAC hardware address.
+    pub ral:                        Volatile<u32>,          // 0x5400
+    /// The higher (most significant) 32 bits of the NIC's MAC hardware address.
+    pub rah:                        Volatile<u32>,          // 0x5404
+    _padding8a:                     [u8; 65536],            // 0x5408 - 0x1FFFF,  109560 bytes
+    _padding8b:                     [u8; 32768],
+    _padding8c:                     [u8;  8192],
+    _padding8d:                     [u8;  2048],
+    _padding8e:                     [u8;   512],
+    _padding8f:                     [u8;   256],
+    _padding8g:                     [u8;   236],
+    _padding8h:                     [u8;    12],
+    // End of all register structs should be at offset 0x20000 (128 KiB in total size).
+
+} // 28 4KiB pages
+
+const_assert_eq!(core::mem::size_of::<E1000MacRegisters>(), 28 * 4096);
```

Comment:
Now that you've split the e1000Registers into 4 separate structs, should we add another const_assert_eq here that checks that all 4 register structs add up to the expected size, which is  defined on Line 124?

Reply:
It would always pass for now, but it doesn't hurt to have more checks. Might be helpful if we ever change how we split up the registers again.